chia/data_layer/data_store.py

```diff
@@ -98,7 +98,6 @@ async def create(cls, db_wrapper: DBWrapper) -> "DataStore":
                     tree_id TEXT NOT NULL,
                     generation INTEGER NOT NULL,
                     PRIMARY KEY(hash, tree_id, generation),
-                    FOREIGN KEY(tree_id, generation) REFERENCES root(tree_id, generation),
```

Comment:
I removed this since the primary key changed, and we'd need a `status` field here as well, which would always have to be equal to `Status.COMMITTED` (since `PENDING` roots never write to ancestor table anymore). I didn't find any way to reference constants as foreign keys (for example `(2, tree_id, generation) REFERENCES root(status, tree_id, generation)`), so our only option would be adding status column and always setting it to `2` constant.